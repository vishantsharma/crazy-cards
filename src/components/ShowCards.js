import React, { useState, useEffect } from 'react';

const ShowCards = props => {
    const [ data, setData ] = useState([]);

    useEffect(() => {
        fetch('http://localhost:3000/data')
        .then(res => res.json())
        .then(responseData => setData(responseData.cards))
    }, [])

    return (
        <ul>
            {data.map(resData => (
                <li> {resData.card_name} </li>
            ))}
            </ul>
    )
}

export default ShowCards;