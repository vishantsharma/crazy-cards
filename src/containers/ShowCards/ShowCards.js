import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import GenerateCards from '../../components/GenerateCards/GenerateCards';
import Card from '../../components/ui/Card/Card';

const ShowCards = () => {
    const [data, setData] = useState([]);
    const [selectedCreditCards, setSelectedCreditCards] = useState({});
    const userDataState = useSelector(state => state.userData);

    useEffect(() => {
        fetch('http://localhost:3000/data')
            .then(res => res.json())
            .then(responseData => setData(responseData.cards))
    }, [])

    const handleClick = cardData => {
        let obj = {};
        obj[cardData.card_type] = cardData.available_credit
        let obj1 = {
            ...obj,
            ...selectedCreditCards
        }
        setSelectedCreditCards(obj1);
        
        if (selectedCreditCards.hasOwnProperty(cardData.card_type)) {
            delete obj1[cardData.card_type]
            setSelectedCreditCards({ ...obj1 });
        }
    }

    const sum = obj => {
        let sum = 0;
        for (let el in obj) {
            if (obj.hasOwnProperty(el)) {
                sum += parseFloat(obj[el]);
            }
        }
        return sum;
    }

    const getFilteredData = (cardsData) => {
        let cardType = [];
        cardsData.forEach(cardData => {
            if ((userDataState.employment === 'student' && cardData.optionsForEmployment === "student")) {
                cardType.push(cardData)
            }
            if ((userDataState.income > 16000 && cardData.income > 16000) && !(cardData.optionsForEmployment === "student")) {
                cardType.push(cardData)
            }
            if (cardData.card_type === "anywhere_card") {
                cardType.push(cardData)
            }
        })
        // eslint-disable-next-line
        return [... new Set(cardType)];
    }

    return (
        <>
            <section className="capture-form">
                <Card>
                    <h2>{`Total Available Credit: Â£${sum(selectedCreditCards)}`}</h2>
                    {getFilteredData(data).map(resData => (
                        <GenerateCards
                            key={resData.card_type}
                            data={resData}
                            handleClick={handleClick} />
                    ))}
                </Card>
            </section>
        </>
    )
}

export default ShowCards;