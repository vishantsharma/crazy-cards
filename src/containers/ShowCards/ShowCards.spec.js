import React from 'react';
import TestRenderer from 'react-test-renderer';
import ShowCards from './ShowCards';

jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    return {
        ...ActualReactRedux,
        useDispatch: jest.fn().mockImplementation(() => {
            return '';
        }),
        useSelector: jest.fn().mockImplementation(() => {
            return '';
        })
    };
});

describe('ShowCards', () => {
    it('should render', () => {
        const component = TestRenderer.create(<ShowCards />)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})