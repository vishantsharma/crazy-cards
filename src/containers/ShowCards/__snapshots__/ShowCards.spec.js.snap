// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCards should render 1`] = `
<section
  className="capture-form"
>
  <div
    className="card"
  >
    <h2>
      Total Available Credit: Â£0
    </h2>
  </div>
</section>
`;
